lib/flutter_sound.dart

```diff
@@ -212,79 +213,82 @@ class FlutterSound {
+          await _channel.invokeMethod('startPlayer', <String, dynamic>{
+        'track': trackMap,
+        'canSkipForward': _skipTrackForwardHandlerSet && canSkipForward,
+        'canSkipBackward': _skipTrackBackwardHandlerSet && canSkipBackward,
+      });
 
-      if (result != null)
-      {
-        print ('startPlayer result: $result');
-        _setPlayerCallback ();
-        _audio_state = t_AUDIO_STATE.IS_PLAYING;
-      }
+      _audio_state = t_AUDIO_STATE.IS_PLAYING;
 
       return result;
     } catch (err) {
       throw Exception(err);
     }
   }
 
-
-  Future<String> startPlayer(String uri) async => _startPlayer('startPlayer', {'path': uri});
-  Future<String> startPlayerFromBuffer(Uint8List dataBuffer) async => _startPlayer('startPlayerFromBuffer', {'dataBuffer': dataBuffer});
-
-
+  /// Stops the media player.
+  ///
+  /// If you would like to continue using the audio player you have to release
+  /// and initialize it again.
   Future<String> stopPlayer() async {
-
-    if (_audio_state != t_AUDIO_STATE.IS_PAUSED && _audio_state != t_AUDIO_STATE.IS_PLAYING ) {
-            throw PlayerRunningException('Player is not playing.');
+    if (_playbackState == null || _playbackState == PlaybackState.STOPPED) {
+      throw PlayerRunningException('Player is not playing.');
     }
 
     _audio_state = t_AUDIO_STATE.IS_STOPPED;
 
-    String result = await _channel.invokeMethod('stopPlayer');
-    _removePlayerCallback();
-    return result;
+    return await _channel.invokeMethod('stopPlayer');
   }
 
   Future<String> pausePlayer() async {
-  if (_audio_state != t_AUDIO_STATE.IS_PLAYING ) {
-          throw PlayerRunningException('Player is not playing.');
-  }
+    if (_playbackState != PlaybackState.PLAYING) {
+      throw PlayerRunningException('Player is not playing.');
+    }
 
-          try {
+    try {
       String result = await _channel.invokeMethod('pausePlayer');
-      if (result != null)
-              _audio_state = t_AUDIO_STATE.IS_PAUSED;
+      if (result != null) _audio_state = t_AUDIO_STATE.IS_PAUSED;
       return result;
     } catch (err) {
       print('err: $err');
-      _audio_state = t_AUDIO_STATE.IS_STOPPED; // In fact _audio_state is in an unknown state
+      _audio_state = t_AUDIO_STATE
+          .IS_STOPPED; // In fact _audio_state is in an unknown state
       return err;
     }
   }
 
   Future<String> resumePlayer() async {
-    if (_audio_state != t_AUDIO_STATE.IS_PAUSED ) {
-          throw PlayerRunningException('Player is not paused.');
+    if (_playbackState != PlaybackState.PAUSED) {
+      throw PlayerRunningException('Player is not paused.');
     }
 
     try {
       String result = await _channel.invokeMethod('resumePlayer');
-      if (result != null)
-              _audio_state = t_AUDIO_STATE.IS_PLAYING;
+      if (result != null) _audio_state = t_AUDIO_STATE.IS_PLAYING;
       return result;
     } catch (err) {
       print('err: $err');
```

Comment:
Same in below. Please remove the `return err`.

Reply:
Same [answer](https://github.com/dooboolab/flutter_sound/pull/202/files/90313028cbfe4c9374627253aaa8445efd983d80#r370956532)